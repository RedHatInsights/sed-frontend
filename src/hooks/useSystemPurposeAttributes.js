import { useQuery } from '@tanstack/react-query';
import useChrome from '@redhat-cloud-services/frontend-components/useChrome';

const fetchSystemPurposeAttributes = (token) => async () => {
  const response = await fetch(
    '/api/rhsm/v2/organization?include=system_purpose_attributes',
    {
      headers: { Authorization: `Bearer ${await token}` },
    }
  );

  const responseData = await response.json();

  return responseData.body;
};

const getSystemPurposeAttributes = (token) => async () => {
  const data = await fetchSystemPurposeAttributes(token)();
  return data.systemPurposeAttributes;
};

const useSystemPurposeAttributes = () => {
  const chrome = useChrome();

  return useQuery(
    ['organization_system_purpose_attributes'],
    getSystemPurposeAttributes(chrome?.auth?.getToken())
  );
};

export { useSystemPurposeAttributes as default };
