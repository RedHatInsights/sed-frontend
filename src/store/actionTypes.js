// Current state actions
export const GET_CURR_STATE = 'GET_CURR_STATE';
export const SET_CURR_STATE = 'SET_CURR_STATE';

// Connected hosts
export const GET_CONNECTED_HOSTS = 'GET_CONNECTED_HOSTS';
