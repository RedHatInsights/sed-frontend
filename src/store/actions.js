import { useConfigApi, useInventoryApi } from '../api';
import {
  GET_CURR_STATE,
  GET_LOG,
  SET_CURR_STATE,
  GET_CONNECTED_HOSTS,
} from './actionTypes';

const fetchCurrState = (api) => () => ({
  type: GET_CURR_STATE,
  payload: api.getProfile('current'),
});

const saveCurrState =
  (api) =>
  ({ compliance, remediations, active }) => ({
    type: SET_CURR_STATE,
    payload: api.createProfile({
      compliance,
      insights: true,
      remediations,
      active,
    }),
  });

const fetchLog =
  (api) =>
  ({ sort = 'created_at:desc', perPage = 50, page = 1 } = {}) => ({
    type: GET_LOG,
    payload: api.getProfiles(sort, perPage, (page - 1) * perPage),
  });

const fetchConnectedHosts = (api) => () => ({
  type: GET_CONNECTED_HOSTS,
  payload: api.getConnectedHosts(),
});

export const useActions = () => {
  const api = useConfigApi();
  const inventoryApi = useInventoryApi();

  return {
    fetchConnectedHosts: fetchConnectedHosts(inventoryApi),
    fetchLog: fetchLog(api),
    saveCurrState: saveCurrState(api),
    fetchCurrState: fetchCurrState(api),
  };
};
