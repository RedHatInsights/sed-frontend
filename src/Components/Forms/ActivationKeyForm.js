import React, { useEffect, useState } from 'react';
import {
  ActionGroup,
  Button,
  Form,
  FormGroup,
  Popover,
  TextInput,
  Text,
  TextContent,
  TextVariants,
} from '@patternfly/react-core';
import HelpIcon from '@patternfly/react-icons/dist/js/icons/help-icon';
import useSystemPurposeAttributes from '../../hooks/useSystemPurposeAttributes';
import ActivationKeysFormSelect from './ActivationKeysFormSelect';
import PropTypes from 'prop-types';
import useNotifications from '../../hooks/useNotifications';

const ActivationKeyForm = (props) => {
  const { handleModalToggle, submitForm, isSuccess, isError, activationKey } =
    props;
  const { addSuccessNotification, addErrorNotification } = useNotifications();
  const { isLoading, error, data } = useSystemPurposeAttributes();
  const [name, setName] = useState('');
  const [role, setRole] = useState('');
  const [serviceLevel, setServiceLevel] = useState('');
  const [usage, setUsage] = useState('');
  const [validated, setValidated] = useState('default');
  const nameRegex = '^[a-zA-Z0-9-_]*$';
  const validationText =
    'Provide a name to be used when registering the activation key. Your activation key name must be unique, may contain only numbers, letters, underscores, and hyphens, and less than 256 characters.';

  const handleSubmit = (event) => {
    event.preventDefault();
    if (validated === 'success' || activationKey) {
      submitForm({
        name: name,
        role: role,
        serviceLevel: serviceLevel,
        usage: usage,
      });
    } else {
      setValidated('error');
    }
  };

  const validateName = (value) => {
    if (value.length === 0 || value.length > 255) {
      setValidated('error');
    } else if (!value.match(nameRegex)) {
      setValidated('error');
    } else {
      setValidated('success');
      setName(value);
    }
  };

  useEffect(() => {
    if (activationKey) {
      setRole(activationKey.role);
      setUsage(activationKey.usage);
      setServiceLevel(activationKey.serviceLevel);
    }
  }, [activationKey]);

  const createButtonDisabled = () => {
    if (activationKey) {
      return (
        activationKey.role === role &&
        activationKey.serviceLevel === serviceLevel &&
        activationKey.usage === usage
      );
    } else {
      return (
        validated === 'error' || name.length === 0 || !name.match(nameRegex)
      );
    }
  };

  if (isSuccess) {
    const successMessage = activationKey
      ? `Activation key ${activationKey.name} updated successfully.`
      : 'Activation key created successfully.';
    addSuccessNotification(successMessage, {
      timeout: false,
    });
    handleModalToggle();
  } else if (isError) {
    const errorMessage = activationKey
      ? `Error updating activation key ${activationKey.name}.`
      : 'Activation Key was not created, please try again.';
    addErrorNotification(errorMessage, {
      timeout: 8000,
    });
    handleModalToggle();
  }

  return (
    <Form id="activation-key-form" onSubmit={handleSubmit}>
      {!activationKey && (
        <FormGroup label="Name" isRequired HelperText={validationText}>
          <TextInput
            id="activation-key-name"
            label="Name"
            isRequired
            type="text"
            validated={validated}
            onChange={(_event, value) => validateName(value)}
            name="name"
          />
        </FormGroup>
      )}
      {activationKey && (
        <FormGroup label="Name">
          {' '}
          <TextContent>{activationKey.name}</TextContent>
        </FormGroup>
      )}
      {!isLoading && !error && (
        <ActivationKeysFormSelect
          data={data.roles}
          value={role}
          onSelect={setRole}
          label="Role"
          name="role"
          placeholderValue="Select role"
          disableDefaultValues={activationKey ? true : false}
          popover={
            <Popover
              bodyContent={
                <TextContent>
                  <Text component={TextVariants.p}>
                    Role is used to categorize systems by the workload on the
                    system
                  </Text>
                  <Text component={TextVariants.p}>
                    Subscription Watch can help you filter and report by these
                    items.
                  </Text>
                  <Text component={TextVariants.p}>
                    Only roles available to your account are shown.
                  </Text>
                </TextContent>
              }
            >
              <HelpIcon />
            </Popover>
          }
          helperText="Select the required role from the list. The list only contains roles available to the activation key."
        />
      )}
      {!isLoading && !error && (
        <ActivationKeysFormSelect
          data={data.serviceLevel}
          value={serviceLevel}
          onSelect={setServiceLevel}
          label="Service Level Agreement (SLA)"
          name="serviceLevel"
          placeholderValue="Select a service level agreement"
          disableDefaultValues={activationKey ? true : false}
          popover={
            <Popover
              bodyContent={
                <TextContent>
                  <Text component={TextVariants.p}>
                    Service Level Agreement (SLA) determines the level of
                    support for systems registered with this activation key.
                  </Text>
                </TextContent>
              }
            >
              <HelpIcon />
            </Popover>
          }
          helperText="Select the required service level from the list. The list only contains service levels available to the activation key."
        />
      )}
      {!isLoading && !error && (
        <ActivationKeysFormSelect
          data={data.usage}
          value={usage}
          onSelect={setUsage}
          label="Usage"
          name="usage"
          placeholderValue="Select usage"
          disableDefaultValues={activationKey ? true : false}
          popover={
            <Popover
              bodyContent={
                <TextContent>
                  <Text component={TextVariants.p}>
                    Usage is used to categorize systems by how they are meant to
                    be used, and therefore supported.
                  </Text>
                  <Text component={TextVariants.p}>
                    Subscription Watch can help you filter and report by these
                    items.
                  </Text>
                </TextContent>
              }
            >
              <HelpIcon />
            </Popover>
          }
          helperText="Select the required usage from the list. The list only contains usages available to the activation key."
        />
      )}
      <ActionGroup>
        <Button
          key="create"
          id="submit-activation-key-button"
          variant="primary"
          type="submit"
          isDisabled={createButtonDisabled()}
          data-testid="activation-key-submit-button"
        >
          {activationKey ? 'Save changes' : 'Create'}
        </Button>

        <Button
          key="cancel"
          id="cancel-activation-key-button"
          variant="link"
          onClick={handleModalToggle}
        >
          Cancel
        </Button>
      </ActionGroup>
    </Form>
  );
};

ActivationKeyForm.propTypes = {
  handleModalToggle: PropTypes.func.isRequired,
  submitForm: PropTypes.func.isRequired,
  isSuccess: PropTypes.bool,
  isError: PropTypes.bool,
  activationKey: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
};

export default ActivationKeyForm;
