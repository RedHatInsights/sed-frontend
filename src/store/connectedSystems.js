import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { GET_CONNECTED_HOSTS } from './actionTypes';

const initialState = {
  loaded: false,
  values: {},
};

const connectedSystemsPending = (state) => ({
  ...state,
  loaded: false,
});

const connectedSystemsFulfilled = (state, { payload }) => ({
  ...state,
  loaded: true,
  hosts: payload?.results || [],
  total: payload?.count || 0,
  page: payload?.page || 0,
  perPage: payload?.per_page || 0,
});

export default applyReducerHash(
  {
    [`${GET_CONNECTED_HOSTS}_PENDING`]: connectedSystemsPending,
    [`${GET_CONNECTED_HOSTS}_FULFILLED`]: connectedSystemsFulfilled,
    [`${GET_CONNECTED_HOSTS}_ERROR`]: () => ({ error: true }),
  },
  initialState
);
