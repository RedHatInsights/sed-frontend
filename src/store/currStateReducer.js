import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { GET_CURR_STATE, SET_CURR_STATE } from './actionTypes';

const initialState = {
  loaded: false,
  values: {},
};

const currStatePending = (state) => ({
  ...state,
  loaded: false,
});
const currStateFulfilled = (state, { payload }) => ({
  ...state,
  loaded: true,
  values: {
    useOpenSCAP: payload?.state?.compliance_openscap === 'enabled',
    enableCloudConnector: payload?.state?.remediations === 'enabled',
    hasInsights: payload?.state?.insights === 'enabled',
  },
});

export default applyReducerHash(
  {
    [`${SET_CURR_STATE}_PENDING`]: currStatePending,
    [`${GET_CURR_STATE}_PENDING`]: currStatePending,
    [`${GET_CURR_STATE}_FULFILLED`]: currStateFulfilled,
    [`${SET_CURR_STATE}_FULFILLED`]: currStateFulfilled,
    [`${GET_CURR_STATE}_ERROR`]: () => ({ error: true }),
  },
  initialState
);
