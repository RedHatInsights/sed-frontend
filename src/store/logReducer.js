import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { GET_LOG } from './actionTypes';

const initialState = {
  loaded: false,
  results: [],
};

const logPending = (state) => ({
  ...state,
  loaded: false,
});
const logFulfilled = (state, { payload }) => ({
  ...state,
  loaded: true,
  ...(payload || {}),
});

export default applyReducerHash(
  {
    [`${GET_LOG}_PENDING`]: logPending,
    [`${GET_LOG}_FULFILLED`]: logFulfilled,
    [`${GET_LOG}_ERROR`]: () => ({ error: true }),
  },
  initialState
);
