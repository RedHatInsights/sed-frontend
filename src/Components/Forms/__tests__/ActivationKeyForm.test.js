import React from 'react';
import ActivationKeyForm from '../ActivationKeyForm';
import useSystemPurposeAttributes from '../../../hooks/useSystemPurposeAttributes';
import { Provider } from 'react-redux';
import { init } from '../../../store';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
const queryClient = new QueryClient();

const handleModalToggle = jest.fn();
const submitForm = jest.fn();

const ActivationKeyFormProps = {
  handleModalToggle,
  submitForm: submitForm,
  isSuccess: null,
  isError: null,
};

const registry = init();
const props = { ...ActivationKeyFormProps };

jest.mock('../../../hooks/useSystemPurposeAttributes');
describe('Activation Key Form', () => {
  beforeEach(() => {
    useSystemPurposeAttributes.mockReturnValue({
      isLoading: false,
      error: false,
      data: {
        roles: ['role'],
        serviceLevel: ['serviceLevel'],
        usage: ['usage'],
      },
    });
  });

  it('renders correctly when data is loaded', () => {
    const { container } = render(
      <Provider store={registry.getStore()}>
        <QueryClientProvider client={queryClient}>
          <ActivationKeyForm {...props} />
        </QueryClientProvider>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('validates name field', () => {
    const props = { ...ActivationKeyFormProps };
    const { container } = render(
      <Provider store={registry.getStore()}>
        <QueryClientProvider client={queryClient}>
          <ActivationKeyForm {...props} />
        </QueryClientProvider>
      </Provider>
    );
    const nameInput = container.querySelector('#activation-key-name');
    fireEvent.change(nameInput, { target: { value: '!123' } });
    expect(nameInput.getAttribute('aria-invalid')).toBe('true');
    fireEvent.change(nameInput, { target: { value: '123Abc#' } });
    expect(nameInput.getAttribute('aria-invalid')).toBe('true');
    fireEvent.change(nameInput, { target: { value: '-Abc_123' } });
    expect(nameInput.getAttribute('aria-invalid')).toBe('false');
  });

  it('validates name length', () => {
    const props = { ...ActivationKeyFormProps };
    const { container } = render(
      <Provider store={registry.getStore()}>
        <QueryClientProvider client={queryClient}>
          <ActivationKeyForm {...props} />
        </QueryClientProvider>
      </Provider>
    );
    const validLength = Array(256).join('a');
    const invalidLength = Array(257).join('b');
    const nameInput = container.querySelector('#activation-key-name');
    fireEvent.change(nameInput, { target: { value: invalidLength } });
    expect(nameInput.getAttribute('aria-invalid')).toBe('true');
    fireEvent.change(nameInput, { target: { value: validLength } });
    expect(nameInput.getAttribute('aria-invalid')).toBe('false');
    fireEvent.change(nameInput, { target: { value: '' } });
    expect(nameInput.getAttribute('aria-invalid')).toBe('true');
  });

  it('disables submit button when editing if no changes were made', () => {
    const activationKey = {
      name: 'test',
      usage: 'test',
      serviceLevel: 'test',
      role: 'test',
    };
    const props = {
      ...ActivationKeyFormProps,
      activationKey: activationKey,
    };
    render(
      <Provider store={registry.getStore()}>
        <QueryClientProvider client={queryClient}>
          <ActivationKeyForm {...props} />
        </QueryClientProvider>
      </Provider>
    );
    const submitButton = screen.getByTestId('activation-key-submit-button');
    expect(submitButton).toBeDisabled();
  });

  it('calls submitForm if form is valid', () => {
    const props = { ...ActivationKeyFormProps };

    const { container } = render(
      <Provider store={registry.getStore()}>
        <QueryClientProvider client={queryClient}>
          <ActivationKeyForm {...props} />
        </QueryClientProvider>
      </Provider>
    );
    const form = container.querySelector('#activation-key-form');
    const nameInput = container.querySelector('#activation-key-name');
    fireEvent.change(nameInput, { target: { value: 'abc-123' } });
    fireEvent.submit(form);
    expect(submitForm).toHaveBeenCalled();
  });

  it('closes the create activation key modal on success', async () => {
    const props = { ...ActivationKeyFormProps, isSuccess: true };

    render(
      <Provider store={registry.getStore()}>
        <QueryClientProvider client={queryClient}>
          <ActivationKeyForm {...props} />
        </QueryClientProvider>
      </Provider>
    );

    expect(handleModalToggle).toHaveBeenCalledTimes(1);
  });
});
