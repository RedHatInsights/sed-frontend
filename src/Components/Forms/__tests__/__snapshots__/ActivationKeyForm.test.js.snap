// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activation Key Form renders correctly when data is loaded 1`] = `
<div>
  <form
    class="pf-v5-c-form"
    id="activation-key-form"
    novalidate=""
  >
    <div
      class="pf-v5-c-form__group"
      helpertext="Provide a name to be used when registering the activation key. Your activation key name must be unique, may contain only numbers, letters, underscores, and hyphens, and less than 256 characters."
    >
      <div
        class="pf-v5-c-form__group-label"
      >
        <label
          class="pf-v5-c-form__label"
        >
          <span
            class="pf-v5-c-form__label-text"
          >
            Name
          </span>
          <span
            aria-hidden="true"
            class="pf-v5-c-form__label-required"
          >
             
            *
          </span>
        </label>
         
      </div>
      <div
        class="pf-v5-c-form__group-control"
      >
        <span
          class="pf-v5-c-form-control"
        >
          <input
            aria-invalid="false"
            data-ouia-component-id="OUIA-Generated-TextInputBase-1"
            data-ouia-component-type="PF5/TextInput"
            data-ouia-safe="true"
            id="activation-key-name"
            label="Name"
            name="name"
            required=""
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
    <div
      class="pf-v5-c-form__group"
      helpertext="Select the required role from the list. The list only contains roles available to the activation key."
    >
      <div
        class="pf-v5-c-form__group-label"
      >
        <label
          class="pf-v5-c-form__label"
        >
          <span
            class="pf-v5-c-form__label-text"
          >
            Role
          </span>
        </label>
         
        <div
          style="display: contents;"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0"
            />
          </svg>
        </div>
      </div>
      <div
        class="pf-v5-c-form__group-control"
      >
        <span
          class="pf-v5-c-form-control"
        >
          <select
            aria-invalid="false"
            aria-label="Select role"
            data-ouia-component-id="OUIA-Generated-FormSelect-default-1"
            data-ouia-component-type="PF5/FormSelect"
            data-ouia-safe="true"
            name="role"
          >
            <option
              class=""
              value="role"
            >
              role
            </option>
            <option
              class=""
              value=""
            >
              Select role
            </option>
          </select>
          <span
            class="pf-v5-c-form-control__utilities"
          >
            <span
              class="pf-v5-c-form-control__toggle-icon"
            >
              <svg
                aria-hidden="true"
                class="pf-v5-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      class="pf-v5-c-form__group"
      helpertext="Select the required service level from the list. The list only contains service levels available to the activation key."
    >
      <div
        class="pf-v5-c-form__group-label"
      >
        <label
          class="pf-v5-c-form__label"
        >
          <span
            class="pf-v5-c-form__label-text"
          >
            Service Level Agreement (SLA)
          </span>
        </label>
         
        <div
          style="display: contents;"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0"
            />
          </svg>
        </div>
      </div>
      <div
        class="pf-v5-c-form__group-control"
      >
        <span
          class="pf-v5-c-form-control"
        >
          <select
            aria-invalid="false"
            aria-label="Select a service level agreement"
            data-ouia-component-id="OUIA-Generated-FormSelect-default-2"
            data-ouia-component-type="PF5/FormSelect"
            data-ouia-safe="true"
            name="serviceLevel"
          >
            <option
              class=""
              value="serviceLevel"
            >
              serviceLevel
            </option>
            <option
              class=""
              value=""
            >
              Select a service level agreement
            </option>
          </select>
          <span
            class="pf-v5-c-form-control__utilities"
          >
            <span
              class="pf-v5-c-form-control__toggle-icon"
            >
              <svg
                aria-hidden="true"
                class="pf-v5-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      class="pf-v5-c-form__group"
      helpertext="Select the required usage from the list. The list only contains usages available to the activation key."
    >
      <div
        class="pf-v5-c-form__group-label"
      >
        <label
          class="pf-v5-c-form__label"
        >
          <span
            class="pf-v5-c-form__label-text"
          >
            Usage
          </span>
        </label>
         
        <div
          style="display: contents;"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0"
            />
          </svg>
        </div>
      </div>
      <div
        class="pf-v5-c-form__group-control"
      >
        <span
          class="pf-v5-c-form-control"
        >
          <select
            aria-invalid="false"
            aria-label="Select usage"
            data-ouia-component-id="OUIA-Generated-FormSelect-default-3"
            data-ouia-component-type="PF5/FormSelect"
            data-ouia-safe="true"
            name="usage"
          >
            <option
              class=""
              value="usage"
            >
              usage
            </option>
            <option
              class=""
              value=""
            >
              Select usage
            </option>
          </select>
          <span
            class="pf-v5-c-form-control__utilities"
          >
            <span
              class="pf-v5-c-form-control__toggle-icon"
            >
              <svg
                aria-hidden="true"
                class="pf-v5-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      class="pf-v5-c-form__group pf-m-action"
    >
      <div
        class="pf-v5-c-form__group-control"
      >
        <div
          class="pf-v5-c-form__actions"
        >
          <button
            aria-disabled="true"
            class="pf-v5-c-button pf-m-primary pf-m-disabled"
            data-ouia-component-id="OUIA-Generated-Button-primary-1"
            data-ouia-component-type="PF5/Button"
            data-ouia-safe="true"
            data-testid="activation-key-submit-button"
            disabled=""
            id="submit-activation-key-button"
            type="submit"
          >
            Create
          </button>
          <button
            aria-disabled="false"
            class="pf-v5-c-button pf-m-link"
            data-ouia-component-id="OUIA-Generated-Button-link-1"
            data-ouia-component-type="PF5/Button"
            data-ouia-safe="true"
            id="cancel-activation-key-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
